src/msdo-client.ts

```diff
@@ -109,36 +109,42 @@ export async function run(inputArgs: string[], telemetryEnvironment: string = 'g
         cliFilePath = process.env.MSDO_FILEPATH;
         core.debug(`cliFilePath = ${cliFilePath}`);
 
-        if (inputArgs != null) {
+        if (inputArgs != null && inputArgs.length != 0) {
             for (let i = 0; i < inputArgs.length; i++) {
                 args.push(inputArgs[i]);
             }
         }
 
-        args.push('--not-break-on-detections');
-
         if (core.isDebug()) {
             args.push('--logger-level');
             args.push('trace');
         }
 
-        let sarifFile : string = path.join(process.env.GITHUB_WORKSPACE, '.gdn', 'msdo.sarif');
-        core.debug(`sarifFile = ${sarifFile}`);
+        let isUploadExisting: boolean = inputArgs[0] == "upload";

```

Comment:
Are we sure that upload will *always* be in this position, or should we consider checking the full array of input args for it?

Reply:
Discussed offline.